presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchTableDescriptionSupplier.java

```diff
@@ -0,0 +1,118 @@
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.nio.file.Files.readAllBytes;
+import static java.util.Objects.requireNonNull;
+
+public class ElasticsearchTableDescriptionSupplier
+        implements Supplier<Map<SchemaTableName, ElasticsearchTableDescription>>
+{
+    private static final Logger log = Logger.get(ElasticsearchTableDescriptionSupplier.class);
+
+    private final JsonCodec<ElasticsearchTableDescription> codec;
+    private final ElasticsearchConnectorConfig config;
+
+    @Inject
+    ElasticsearchTableDescriptionSupplier(ElasticsearchConnectorConfig config, JsonCodec<ElasticsearchTableDescription> codec)
+    {
+        this.codec = requireNonNull(codec, "codec is null");
+        this.config = requireNonNull(config, "config is null");
+    }
+
+    @Override
+    public Map<SchemaTableName, ElasticsearchTableDescription> get()
+    {
+        return createElasticsearchTableDescriptions(config.getTableDescriptionDir(), config.getDefaultSchema(), ImmutableSet.copyOf(config.getTableNames()), codec);
+    }
+
+    public static Map<SchemaTableName, ElasticsearchTableDescription> createElasticsearchTableDescriptions(File tableDescriptionDir, String defaultSchema, Set<String> tableNames, JsonCodec<ElasticsearchTableDescription> tableDescriptionCodec)
+    {
+        ImmutableMap.Builder<SchemaTableName, ElasticsearchTableDescription> builder = ImmutableMap.builder();
+
+        log.debug("Loading elasticsearch table definitions from %s", tableDescriptionDir.getAbsolutePath());
+
+        try {
+            for (File file : listFiles(tableDescriptionDir)) {
+                if (file.isFile() && file.getName().endsWith(".json")) {
+                    ElasticsearchTableDescription table = tableDescriptionCodec.fromJson(readAllBytes(file.toPath()));
+                    String schemaName = firstNonNull(table.getSchemaName(), defaultSchema);
+                    log.debug("Elasticsearch table %s.%s: %s", schemaName, table.getTableName(), table);
+                    builder.put(new SchemaTableName(schemaName, table.getTableName()), table);
+                }
+            }
+
+            Map<SchemaTableName, ElasticsearchTableDescription> tableDefinitions = builder.build();
+
+            log.debug("Loaded Table definitions: %s", tableDefinitions.keySet());
+
+            builder = ImmutableMap.builder();
+            for (String definedTable : tableNames) {
+                SchemaTableName tableName;
+                tableName = parseTableName(definedTable);
+
+                if (!tableDefinitions.containsKey(tableName)) {
+                    throw new IOException("Missing table definition for: " + tableName);
```

Comment:
`IOException` doesn't really fit here. You can throw a `PrestoException` instead.